frontend/src/components/pages/FilterBox.tsx

```diff
@@ -31,54 +43,132 @@ const FilterBox = ({
     setGenreFilter([]);
     setAgeFilter([]);
   };
+
+  const clearTag = async (type: any, val: any) => {
+    if (type === "genre") {
```

Comment:
We should definitely define an enum for tag types, define an enum in the types.ts file for genre and age and then set the type of the type parameter to `type: newEnum`.